feat: aws-kms submodule example (#3)
terraform-mongodbatlas-modules/terraform-mongodbatlas-encryption-at-rest: examples/aws-kms-key-complete/README.md

# encryption-at-rest-aws-kms-key - complete example

_Note: you can see the full source code in the [github repository](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-encryption-at-rest/tree/main/examples/aws-kms-key-complete)_

> This example sets up encryption at rest using an AWS KMS for your Atlas Project. It creates the encryption key in AWS KMS, an IAM role and policy so that Atlas can access the key, and enables encryption at rest for the Atlas Project. Finally, it creates a Cluster with encryption at rest enabled.

Reviewer: Do you think this would be better as a bullet or numbered list similar to the following: 
```suggestion
This example sets up encryption at rest using an AWS KMS for your Atlas Project. Specifically, it does the following: 

- Creates encryption key in AWS KMS.
- Creates an IAM role and policy so that Atlas can access the key.
- Enables encryption at rest for the Atlas Project.
- Creates a Cluster with encryption at rest enabled.
```

## Usage

- Set the following variable: 

>     - `project_id`: ID of the Atlas project

Reviewer: ```suggestion
    - `project_id`: Unique 24-hexadecimal character string that identifies the Atlas project
```

- Set the following environment variables:

    -  `export MONGODB_ATLAS_PUBLIC_KEY="<YOUR_PUBLIC_KEY>"`
    -  `export MONGODB_ATLAS_PRIVATE_KEY="<YOUR_PRIVATE_KEY>"`
    -  `export AWS_ACCESS_KEY_ID="<YOUR_ACCESS_KEY>"`
    -  `export AWS_SECRET_ACCESS_KEY="<YOUR_SECRET_KEY>"`
    -  `export AWS_REGION="<YOUR_REGION>"`

- Run the following command to initialize your project:

```bash
$ terraform init
```

- Run the following command to review the execution plan:

```bash
$ terraform plan
```

- Run the following command to deploy your infrastructure:

```bash
$ terraform apply
```

> ## Considerations

Reviewer: Should this be Requirements or Pre-Requisites?

> - In order to define the AWS KMS Key policy with the correct permissions, it should look like this:

Reviewer: ```suggestion
- Your AWS KMS Key policy configuration with the correct permissions must resemble the following:
```

> 

Reviewer 2: I think it is a bit strong wording. Maybe instead:
Your AWS KMS Key policy must allow the IAM Role access, this can be done by:

```terraform
{
  "Statement": [
      // other statements
      {
          "Sid": "Allow use of the key by specific IAM user",
          "Action": [
              "kms:Decrypt",
              "kms:Encrypt",
              "kms:DescribeKey"
          ],
          "Effect": "Allow",
          "Principal": {
              "AWS": "*"
          },
          "Resource": "*",
          "Condition": {
              "StringEquals": {

>                   "aws:PrincipalArn": "arn:aws:iam::{ACCOUNT}:role/IAM_EXECUTION_ROLE" // optional clause to limit to your execution role only

Reviewer 2: Remove the comment (//optional clause.... )
Otherwise, the permission is too open

# encryption-at-rest-aws-kms-key - complete example

_Note: you can see the full source code in the [github repository](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-encryption-at-rest/tree/main/examples/aws-kms-key-complete)_

This example sets up encryption at rest using an AWS KMS for your Atlas Project. Specifically, it does the following: 
- Creates encryption key in AWS KMS.
- Creates an IAM role and policy so that Atlas can access the key.
- Enables encryption at rest for the Atlas Project.

> - Creates a Cluster with encryption at rest enabled.

Reviewer 3: Could we clarify which resources are created from the module? Specifically:

> - Creates an IAM role and policy so that Atlas can access the key.
> - Enables encryption at rest for the Atlas Project.

## Usage

- Set the following variable: 

    - `project_id`: Unique 24-hexadecimal character string that identifies the Atlas project

- Set the following environment variables:

    -  `export MONGODB_ATLAS_PUBLIC_KEY="<YOUR_PUBLIC_KEY>"`
    -  `export MONGODB_ATLAS_PRIVATE_KEY="<YOUR_PRIVATE_KEY>"`
    -  `export AWS_ACCESS_KEY_ID="<YOUR_ACCESS_KEY>"`
    -  `export AWS_SECRET_ACCESS_KEY="<YOUR_SECRET_KEY>"`
    -  `export AWS_REGION="<YOUR_REGION>"`

- Run the following command to initialize your project:

```bash
$ terraform init
```

- Run the following command to review the execution plan:

```bash
$ terraform plan
```

- Run the following command to deploy your infrastructure:

```bash
$ terraform apply
```

## Considerations

> - Your AWS KMS Key policy must allow the IAM Role access, this can be done by::

Reviewer 3: ```suggestion
- Your AWS KMS Key policy must allow the IAM Role access. The following is an example of how to provide this permission:
```
question: is the statement "must" accurate here? Asking because in our terraform provider examples [the kms key does not define any policy](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/mongodbatlas_encryption_at_rest/aws/atlas-cluster/aws-kms.tf#L1). 

Also would define the kms key policy in the actual example when creating the kms key if we are giving the suggestion.

Author: yes, maybe we can redact it with "should" instead. Will add the policy to the example
